check.py

```diff
@@ -47,10 +52,33 @@ def warn(text):
 
 # setup
 
-BASE_DIR = os.path.abspath('test')
-WATERFALL_BUILD_DIR = os.path.join(BASE_DIR, 'wasm-install')
+# Locate Binaryen build artifacts directory (bin/ by default)
+if not options.binaryen_bin:
+  if os.environ.get('BINARYEN_ROOT'):
+    if os.path.isdir(os.path.join(os.environ.get('BINARYEN_ROOT'), 'bin')): options.binaryen_bin = os.path.join(os.environ.get('BINARYEN_ROOT'), 'bin')
+    else: options.binaryen_bin = os.environ.get('BINARYEN_ROOT')
+  else:
+    options.binaryen_bin = 'bin'
+
+if not os.path.isfile(os.path.join(options.binaryen_bin, 'wasm-dis')) and not os.path.isfile(os.path.join(options.binaryen_bin, 'wasm-dis.exe')):
```

Comment:
Seems like we could factor this kind of thing into some function like `ExecutableName()` which might append `.exe` on Windows and nothing otherwise?